lib/floe/workflow/context.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Floe
+  class Workflow
+    class Context
+      def initialize(context = nil, input: {})
+        context = JSON.parse(context) if context&.kind_of?(String)
+
+        @context = context || {
+          "Execution"    => {
+            "Input" => input
+          },
+          "State"        => {},
+          "States"       => [],
+          "StateMachine" => {
+          },
+          "Task"         => {
+          }
+        }
+      end
+
+      def [](key)
+        @context[key]
+      end
+
+      def []=(key, val)
+        @context[key] = val
+      end
+
+      def dig(*args)
+        @context.dig(*args)
+      end
```

Comment:
I don't want this to just be free-for-all access to the `@context` object, maybe accessor functions like `#state` and `#execution` but we still need JsonPath access for e.g. `$$.State.EnteredTime`